### PR TITLE
Adding error assertions check

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: errorAssertions/errorAssertions.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch x := node.(type) {
+			case *ast.CallExpr:
+				callExpr, _ := node.(*ast.CallExpr)
+				fun, ok := x.Fun.(*ast.SelectorExpr)
+				if !ok {
+					return true
+				}
+
+				module, ok := fun.X.(*ast.Ident)
```
Comment (agnivade) [inline errorAssertions/errorAssertions.go]: This can be named as "idnt"

File: errorAssertions/errorAssertions.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+				}
+
+				if fun.Sel.Name == "NotNil" && (module.Name == "require" || module.Name == "assert") && len(callExpr.Args) > 1 {
+					if typeAndValue, ok := pass.TypesInfo.Types[callExpr.Args[1]]; ok && typeAndValue.Type.String() == "error" {
+						pass.Reportf(callExpr.Pos(), "calling require/assert.NotNil on a regular error, please use require/assert.Error instead")
+					}
+				}
+
+				if fun.Sel.Name == "Error" && (module.Name == "require" || module.Name == "assert") && len(callExpr.Args) > 1 {
+					if typeAndValue, ok := pass.TypesInfo.Types[callExpr.Args[1]]; ok && typeAndValue.Type.String() == "*github.com/mattermost/mattermost-server/v5/model.AppError" {
```
Comment (agnivade) [inline errorAssertions/errorAssertions.go]: I'd refactor `*github.com/mattermost/mattermost-server/v5/model.AppError` to be a const

File: errorAssertions/errorAssertions.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				if !ok {
+					return true
+				}
+
+				module, ok := fun.X.(*ast.Ident)
+				if !ok {
+					return true
+				}
+
+				if fun.Sel.Name == "Nil" && (module.Name == "require" || module.Name == "assert") && len(callExpr.Args) > 1 {
```
Comment (agnivade) [inline errorAssertions/errorAssertions.go]: I think these four if conditions could be run in a single loop with the variable type and the call details being elements of a slice.

And we can return false whenever we have found a matching condition.

**Review by agnivade — CHANGES_REQUESTED** [review]
Some suggestions to refactor the code. Looking great otherwise.

File: errorAssertions/errorAssertions.go (hunk 2 above)
Comment (jespino) [reply errorAssertions/errorAssertions.go]: done

File: errorAssertions/errorAssertions.go (hunk 3 above)
Comment (jespino) [reply errorAssertions/errorAssertions.go]: done

File: errorAssertions/errorAssertions.go (hunk 1 above)
Comment (jespino) [reply errorAssertions/errorAssertions.go]: done

File: errorAssertions/errorAssertions.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+				idnt, ok := fun.X.(*ast.Ident)
+				if !ok {
+					return true
+				}
+
+				if (idnt.Name == "require" || idnt.Name == "assert") && len(callExpr.Args) > 1 {
+					for _, check := range checks {
+						if fun.Sel.Name == check.AssertionName {
+							if typeAndValue, ok := pass.TypesInfo.Types[callExpr.Args[1]]; ok && typeAndValue.Type.String() == check.ForbiddenTypeName {
+								pass.Reportf(callExpr.Pos(), "calling %s.%s on %s, please use %s.%s instead", idnt.Name, check.AssertionName, check.ForbiddenTypeName, idnt.Name, check.ExpectedAssertion)
```
Comment (agnivade) [inline errorAssertions/errorAssertions.go:53]: We can return false once we have found a match.

**Review by agnivade — APPROVED** [review]
Looks great!

File: errorAssertions/errorAssertions.go (hunk 4 above)
Comment (jespino) [reply errorAssertions/errorAssertions.go:53]: done

**Pull request merged**